src/lib/config-utils.ts

```diff
@@ -0,0 +1,376 @@
+import * as Promise from 'bluebird';
+import * as childProcessSync from 'child_process';
+import * as _ from 'lodash';
+import { fs } from 'mz';
+
+import * as constants from './constants';
+import * as fsUtils from './fs-utils';
+import { EnvVarObject } from './types';
+
+const childProcess: any = Promise.promisifyAll(childProcessSync);
+
+export interface ConfigOptions {
+	[key: string]: string | string[];
+}
+
+export interface ExtLinuxFile {
+	labels: {
+		[labelName: string]: {
+			[directive: string]: string;
+		};
+	};
+	globals: { [directive: string]: string };
+}
+
+export const rpiArrayConfigKeys = [
+	'dtparam',
+	'dtoverlay',
+	'device_tree_param',
+	'device_tree_overlay',
+];
+
+export const extlinuxSupportedConfig = [
+	'isolcpus',
+];
+
+export const hostConfigConfigVarPrefix = 'RESIN_HOST_';
+export const rPiBootConfigPrefix = hostConfigConfigVarPrefix + 'CONFIG_';
+export const extlinuxBootConfigPrefix = hostConfigConfigVarPrefix + 'EXTLINUX_';
+export const rPiConfigRegex = new RegExp('(' + _.escapeRegExp(rPiBootConfigPrefix) + ')(.+)');
+export const extlinuxConfigRegex = new RegExp('(' + _.escapeRegExp(extlinuxBootConfigPrefix) + ')(.+)');
+
+export const bootMountPoint = constants.rootMountPoint + constants.bootMountPoint;
+
+export function isRPiDeviceType(deviceType: string): boolean {
+	return _.startsWith(deviceType, 'raspberry') || deviceType === 'fincm3';
+}
+
+export function isExtlinuxDeviceType(deviceType: string): boolean {
+	return _.startsWith(deviceType, 'jetson-tx');
+}
+
+export function isConfigDeviceType(deviceType: string): boolean {
+	return isRPiDeviceType(deviceType) || isExtlinuxDeviceType(deviceType);
+}
+
+export function bootConfigVarPrefix(deviceType: string): string {
+	if (isRPiDeviceType(deviceType)) {
+		return rPiBootConfigPrefix;
+	}	else if (isExtlinuxDeviceType(deviceType)) {
+		return extlinuxBootConfigPrefix;
+	} else {
+		throw new Error(`No boot config var prefix for device type: ${deviceType}`);
+	}
+}
+
+export function bootConfigVarRegex(deviceType: string): RegExp {
+	if (isRPiDeviceType(deviceType)) {
+		return rPiConfigRegex;
+	} else if (isExtlinuxDeviceType(deviceType)) {
+		return extlinuxConfigRegex;
+	} else {
+		throw new Error(`No boot config var regex for device type: ${deviceType}`);
+	}
+}
+
+export function envToBootConfig(deviceType: string, env: EnvVarObject): ConfigOptions {
+	if (!isConfigDeviceType(deviceType)) {
+		return { };
+	}
+	const prefix = bootConfigVarPrefix(deviceType);
+	const regex = bootConfigVarRegex(deviceType);
+
+	let parsedEnv = _.pickBy(env, (_val: string, key: string) => {
+		return _.startsWith(key, prefix);
+	});
+	parsedEnv = _.mapKeys(parsedEnv, (_val, key) => {
```

Comment:
_.mapKeys takes a callback that receives the key as the first argument, right?

Reply:
Doesn't seem to: https://lodash.com/docs/4.17.10#mapKeys